spec/unit/log_spec.rb

```diff
@@ -200,4 +200,18 @@
       end
     }.to raise_error(TTY::Logger::Error, "Already defined log type :success")
   end
+
+  it "logs streaming output" do
+    logger = TTY::Logger.new(output: output) do |config|
+      config.level = :debug
```

Comment:
I'd expect this configuration to be reflected in the logging output for the `<<` call. Wouldn't you?

Reply:
No, I disagree here. Making this change overloads the meaning of `config.level`, which is
<em>let's log all events that have a `:debug` level or higher</em>
Your requested change adds:
<em>let's log all output **assigning** them `:debug` level</em>
If someone changes the level, the output level will change too, and this makes logging unpredictable. (If I set, e.g. `:warning`, I would be asking to skip all logging except errors, for instance.)

Reply:
In my mind, the exact meaning of the `<<` or `write` is to dump any output to the logger irrespective of the current logging level. That's exactly how the Ruby built-in `Logger` works. Irrespective of set logger level it logs everything when the `<<` is called on it. Why? The writing caller, like a long-running process, doesn't know what is an error or what is an info message. Consequently, it cannot select an appropriate level so it needs to write everything.  I have changed [tty-command logger example](https://github.com/piotrmurach/tty-command/blob/master/examples/logger.rb) to demonstrate that setting log level has no impact. 

Conversely, if we were to make the change to `:debug` level for `<<` as suggested in the code. Then, if you were to create `TTY::Logger` instance with `:info` level, the external process would never be able to write anything to the logger. Why? Because the logger expects `:info` or higher but all it gets is `:debug` level messages. If you change the test to `:info` level you will get an empty string. In practice, if you had logger set on anything than `:debug` level you would never see any messages written including some serious errors your running process is trying to send. To me this is rather unexpected and erroneous behaviour that may potentially lead someone to believe that their external process is behaving fine based on the logs inspection. 

Changing the `<<` to the current level will ensure that all messages are captured and they are printed matching the level set. Ideally, they wouldn't be printed with any level since we cannot infer this from the sent messages.

Reply:
@amyspark ping